### DISCUSSION TITLE
Restructured Text Rendering is broken

### DISCUSSION THREAD

**Discussion by rhpvorderman** [community/community#114690] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I have several projects that have badges on the README. The README is in Restructured Text format.

The badges used to be rendered like this:

```
[Badge1][Badge2][Badge3][Badge4][Badge5][Badge6]
[Badge7][Badge etc.]

PROJECT TITLE
```
As much images as possible would be fitted side by side. Now they are rendered as follows:

```
[Badge1]

[Badge2]

[Badge3]

[Badge4]

[Badge5]

[Badge6]

[Badge7]

[Badge etc.]

PROJECT TITLE
```

Has the Restrtuctured Text renderer changed? 

The restructured text code used is:
```REST

.. image:: https://img.shields.io/pypi/v/sequali.svg
  :target: https://pypi.org/project/sequali/
  :alt:

.. image:: https://img.shields.io/conda/v/bioconda/sequali.svg
  :target: https://bioconda.github.io/recipes/sequali/README.html
  :alt:

.. image:: https://img.shields.io/pypi/pyversions/sequali.svg
  :target: https://pypi.org/project/sequali/
  :alt:

.. image:: https://img.shields.io/pypi/l/sequali.svg
  :target: https://github.com/rhpvorderman/sequali/blob/main/LICENSE
  :alt:

.. image:: https://readthedocs.org/projects/sequali/badge/?version=latest
  :target: https://sequali.readthedocs.io/en/latest/?badge=latest
  :alt:

.. image:: https://codecov.io/gh/rhpvorderman/sequali/graph/badge.svg?token=MSR1A6BEGC
  :target: https://codecov.io/gh/rhpvorderman/sequali
  :alt:

```

EDIT: Also  code rendering is broken. My `code-block:: yaml` is now rendered with only underlining. The same problem is also posted by @jonas-eschle .

**Comment by jonas-eschle** [comment]
I see the same for my repos. Furthermore, the rendering of code seems to have changed, instead of highlighting it, it's just underlined and all in black

**Reply by rhpvorderman** [reply]
@jonas-eschle I have the same issue. Still the issue isn't noticed. I will change the title in a fashion that asks more attention.

**Reply by rhpvorderman** [reply]
Right, I called it "restructured text rendering is broken". This is the truth and should capture the attention of those who can fix it. Hopefully. Also this comment bumps it to the top.

**Comment by RobPasMue** [comment]
I opened a GitHub support ticket recently with this same issue. Apparently they are investigating it - but no fix yet. This has been going on for weeks already. See https://github.com/ansys/pyansys-geometry

**Reply by RobPasMue** [reply]
It affects mostly code blocks on our side - just like @jonas-eschle  mentioned

**Comment by abravalheri** [comment]
I created a related issue recently in https://github.com/github/markup/issues/1801#issue-2252314843.
I am reporting the content below for the sake of contributing to the problem description:

---

It is a common pattern in the Python community to use something similar to the following snippet (in ReStructuredText) to add "badges" as a UI block at the top of the README of repositories:

```rst
.. image:: https://img.shields.io/pypi/v/setuptools.svg
   :target: https://pypi.org/project/setuptools

.. image:: https://img.shields.io/pypi/pyversions/setuptools.svg

.. image:: https://github.com/pypa/setuptools/actions/workflows/main.yml/badge.svg
   :target: https://github.com/pypa/setuptools/actions?query=workflow%3A%22tests%22
   :alt: tests

.. image:: https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v2.json
    :target: https://github.com/astral-sh/ruff
    :alt: Ruff

.. image:: https://img.shields.io/readthedocs/setuptools/latest.svg
    :target: https://setuptools.pypa.io

.. image:: https://img.shields.io/badge/skeleton-2024-informational
   :target: https://blog.jaraco.com/skeleton

.. image:: https://img.shields.io/codecov/c/github/pypa/setuptools/master.svg?logo=codecov&logoColor=white
   :target: https://codecov.io/gh/pypa/setuptools

.. image:: https://tidelift.com/badges/github/pypa/setuptools?style=flat
   :target: https://tidelift.com/subscription/pkg/pypi-setuptools?utm_source=pypi-setuptools&utm_medium=readme

.. image:: https://img.shields.io/discord/803025117553754132
   :target: https://discord.com/channels/803025117553754132/815945031150993468
   :alt: Discord
```

Previously, that would render as a horizontal bar:

![before](https://github.com/github/markup/assets/320755/d204141f-2289-483d-bec8-e0639e796ca6)

But now each image is occupying a new line:

![after](https://github.com/github/markup/assets/320755/5aa8ec74-3ceb-4742-b2bb-516f17d3b5e9)

Please note that this was a pattern that historically emerged in the community, so it is wide spread.
Therefore this change affects a lot of repositories, and in my opinion represents a regression.

Examples of repositories affected by this change:

https://github.com/python/cpython/tree/v3.13.0a6
https://github.com/pypa/setuptools/tree/v69.5.1
https://github.com/pyca/cryptography/tree/42.0.5
https://github.com/pytest-dev/pytest/tree/v7.4.3
https://github.com/benjaminp/six/tree/1.16.0
https://github.com/scipy/scipy/tree/v1.13.0
https://github.com/pyscaffold/pyscaffold/tree/v4.5
https://github.com/pyca/pynacl/tree/v0.3.0
https://github.com/jpadilla/pyjwt/tree/2.8.0
https://github.com/jaraco/zipp/tree/v3.18.1
https://github.com/python/importlib_metadata/tree/v7.1.0
https://github.com/python/importlib_resources/tree/v6.4.0
https://github.com/pypa/readme_renderer/tree/43.0

and many more ...

**Reply by rhpvorderman** [reply]
Ah, thanks. That was the repo I should have complained in originally. Thanks for pointing out where to look!

**Comment by jonas-eschle** [comment]
Thanks to everyone in this discussion: I've reached out to the support a while ago and they said that they're tracking it internally, they seem to be aware of the issue.

**Comment by shenxianpeng** [comment]
A temporary solution could be to use Restructured Text substitution to put badges in a line. see this PR https://github.com/pypa/setuptools/pull/4312, but it's best that GitHub fixes this.

**Comment by adamchainz** [comment]
I’m seeing further breakage today with links using `replace::` no longer working. See [django-browser-reload](https://github.com/adamchainz/django-browser-reload) README “Requirements” section.

Broken on GitHub:

![Screenshot 2024-04-23 at 15-55-24 adamchainz_django-browser-reload Automatically reload your browser in development](https://github.com/community/community/assets/857609/feffe2ea-eb4b-4ed4-8990-9ff105bc6759)

Correct on PyPI, and how it used to render on GitHub:

![Screenshot 2024-04-23 at 15-55-05 django-browser-reload](https://github.com/community/community/assets/857609/70555d7a-57ff-471a-aa7d-9dc8cafec6f5)

**Comment by jwodder** [comment]
Two more problems with reStructuredText rendering that haven't been mentioned yet:

* [Doctest blocks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#doctest-blocks) are no longer being properly rendered.  For example, the following markup:

    ```rst
    Condense commit years obtained from ``git log`` or the like into ``range``
    objects:

    >>> import derange
    >>> derange.derange([2015, 2015, 2015, 2014, 2014, 2011, 2010, 2010, 2009, 2009])
    [range(2009, 2012), range(2014, 2016)]
    ```

    should be rendered like (screenshot taken from PyPI):

    <img width="803" alt="Screenshot 2024-04-23 at 19 13 32" src="https://github.com/community/community/assets/98207/26dd2f97-480f-421e-b93a-1687472ba44c">

    but is currently rendering as:

    <img width="772" alt="Screenshot 2024-04-23 at 19 13 53" src="https://github.com/community/community/assets/98207/11ca14a6-dead-4b53-970e-a8fe10b7ce51">

* [Option lists](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#option-lists) are no longer being properly rendered.  For example, the following markup:

    ```rst
    Options
    -------

    -J, --json                  Output JSON

    -r REMOTE, --remote REMOTE  Parse the GitHub URL from the given remote
                                [default: origin]
    ```

    should be rendered like (screenshot taken from PyPI):

    <img width="436" alt="Screenshot 2024-04-23 at 19 16 33" src="https://github.com/community/community/assets/98207/24e2ee4a-0e90-4290-9649-551df8a857f8">

    but is currently rendering as:

    <img width="586" alt="Screenshot 2024-04-23 at 19 16 47" src="https://github.com/community/community/assets/98207/05e35cda-914e-4edf-ad33-0a375ea97dac">
